packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/node/ImageNode.tsx

```diff
@@ -16,40 +16,53 @@ import { Theme, useTheme } from '@material-ui/core/styles';
 import { memo, useContext } from 'react';
 import { Handle, NodeProps, NodeResizer, Position } from 'reactflow';
 import { Label } from '../Label';
+import { DropNodeContext } from '../dropNode/DropNodeContext';
+import { DropNodeContextValue } from '../dropNode/DropNodeContext.types';
 import { NodePalette } from '../palette/NodePalette';
 import { ImageNodeData } from './ImageNode.types';
 
 const imageNodeStyle = (
   theme: Theme,
   style: React.CSSProperties,
   selected: boolean,
-  faded: boolean
+  faded: boolean,
+  compatibleDropTarget: boolean,
+  dropTarget: boolean
 ): React.CSSProperties => {
   const imageNodeStyle: React.CSSProperties = { width: '100%', height: '100%', opacity: faded ? '0.4' : '', ...style };
+
   if (selected) {
     imageNodeStyle.outline = `${theme.palette.primary.main} solid 1px`;
+  } else if (dropTarget && compatibleDropTarget) {
+    imageNodeStyle.boxShadow = `0px 0px 2px 2px ${theme.palette.primary.main}`;
   }
 
   return imageNodeStyle;
 };
 
 export const ImageNode = memo(({ data, isConnectable, id, selected }: NodeProps<ImageNodeData>) => {
   const theme = useTheme();
+  const { dropData } = useContext<DropNodeContextValue>(DropNodeContext);
+  const isCompatibleDropTarget: boolean = dropData.compatibleNodeIds.includes(id);
+  const isSelectedDropTarget: boolean = id === dropData.targetNodeId;
+  const faded =
+    data.faded || (dropData.draggedNodeId !== null && dropData.draggedNodeId !== id && !isCompatibleDropTarget);
```

Comment:
All this should be in the useDropNode hook to provide something like this:

```
const { nodeStyleProvider } = useDropNode()
const dropStyle: React.CssProperties = nodeStyleProvider.getStyle(id);
```

and then

```
style={...imageNodeStyle(theme, data.style, selected, data.faded), ...dropStyle}
```

I don't need any more impact than that on the various nodes (and the future custom nodes)

Reply:
Yes, the API needed was not clear yet, hence the draft status.
I'll move the "gory details" about the data structures manipulated into the hook and only expose a simple API for users.